### PR TITLE
Use service Type=notify and generate swap.conf

### PR DESCRIPTION
**Pull request by zenofile**
Tries to communicate the current state to systemd.

### CODE REVIEW

**Comment by zenofile** [comment]
Missing comments on swap.conf - fix soon.

File: systemd-swap.service [hunk 1]
```diff
@@ -2,6 +2,7 @@
 Description=Manage swap spaces on zram, files and partitions.
 
 [Service]
+Type=notify
```
Comment (Jarel1337) [inline systemd-swap.service:5]: 👍

File: Makefile [hunk 2]
```diff
@@ -30,6 +29,12 @@ $(CNF_T): swap.conf
 
 files: $(BIN_T) $(SVC_T) $(DFL_T) $(CNF_T)
 
+define banner
+#  This file is part of systemd-swap.\n#\n# Entries in this file show the systemd-swap defaults as\n# specified in /usr/share/systemd-swap/swap-default.conf\n# You can change settings by editing this file.\n# Defaults can be restored by simply deleting this file.\n#\n# See swap.conf(5) and /usr/share/systemd-swap/swap-default.conf for details.\n
+endef
+swap.conf:
+	echo -e "$(banner)" > $@
+	grep -o '^[^#]*' swap-default.conf | sed 's/^/#/' >> $@
```
Comment (Jarel1337) [inline Makefile:37]: 👍 Makes things less confusing (imo)

File: systemd-swap [hunk 3]
```diff
@@ -315,6 +320,7 @@ case "$1" in
 
       to_bytes(){ numfmt --to=none --from=iec "$1"; }
 
+      systemd-notify "STATUS=Monitoring memory status..."
```
[the remark this answers is not in the document]
Comment (zenofile) [reply systemd-swap:323]: Yes. I am trying to think of an easy solution for this, without hammering out systemd-notify calls.

[the remark this answers is not in the document]
Comment (zenofile) [reply systemd-swap:323]: I guess I could just call `systemd-notify "STATUS=Monitoring memory status..."` after allocating or deallocating. Looks ugly, but works.

**Comment by Jarel1337** [comment]
Wanted to squash your commits into two but in doing so I realized that I couldn't push those changes to your branch (no permission) I did however do so locally, so the git history won't look any different. The only difference is that this pull request now looks like it's still open even though it's merged, so I'll go ahead and close it. Commit 0d955f324edab93dddb429b3bb26a35d9a981d21 merged this branch.

**Comment by zenofile** [comment]
Please see https://github.com/Nefelim4ag/systemd-swap/commit/3aad6df77fd99ff71f5705af900cbf11d17d30c1#r39869034.

**Pull request closed without merging**